packages/braid-design-system/src/lib/components/private/Popover/Popover.tsx

```diff
@@ -0,0 +1,360 @@
+      }
+
+      const popoverBoundingRect = ref?.current?.getBoundingClientRect();
+      if (!popoverBoundingRect) {
+        return;
+      }
+
+      const { width: popoverWidth } = popoverBoundingRect;
+
+      const triggerCenter =
+        triggerPosition.width &&
+        triggerPosition.left + triggerPosition.width / 2;
+
+      const popoverLeft =
+        alignProp === 'center' && triggerCenter
+          ? triggerCenter - popoverWidth / 2
+          : triggerPosition.left;
+
+      const clampedPopoverLeft = clamp(
+        scrollX,
+        popoverLeft,
+        window.innerWidth + scrollX - popoverWidth,
+      );
+
+      const triggerRightFromLeft = window.innerWidth - triggerPosition.right;
+
+      const clampedTriggerRightFromLeft = clamp(
+        scrollX + popoverWidth,
+        triggerRightFromLeft,
+        scrollX + window.innerWidth,
+      );
+
+      if (
+        align === 'right' &&
+        clampedTriggerRightFromLeft !== triggerPosition.right + horizontalOffset
+      ) {
+        setHorizontalOffset(
+          window.innerWidth -
+            clampedTriggerRightFromLeft -
+            triggerPosition.right,
+        );
+      }
+      if (
+        align === 'left' &&
+        clampedPopoverLeft !== triggerPosition.left + horizontalOffset
+      ) {
+        setHorizontalOffset(clampedPopoverLeft - triggerPosition.left);
+      }
+
+      return;
+    };
+
+    useIsomorphicLayoutEffect(() => {
+      if (!showPopover) {
+        return;
+      }
+
+      if (width !== 'full') {
+        handleHorizontalShift();
+      }
+      if (!lockPlacement) {
+        handlePlacement();
+      }
+    });
+
+    const triggerPositionVars = triggerPosition && {
+      // Vertical positioning
+      [styles.triggerVars[actualPlacement]]:
+        `${triggerPosition[actualPlacement]}`,
+
+      // Horizontal positioning
+      [styles.triggerVars.left]:
+        width === 'full' || align === 'left'
+          ? `${triggerPosition?.left}`
+          : undefined,
+      [styles.triggerVars.right]:
+        width === 'full' || align === 'right'
+          ? `${triggerPosition?.right}`
+          : undefined,
+
+      // Horizontal scroll offset
+      [styles.horizontalOffset]: width !== 'full' ? `${horizontalOffset}` : '0',
+    };
+
+    if (!showPopover) {
+      return null;
+    }
+
+    return (
+      <BraidPortal>
+        <Box
+          data-testid={
+            process.env.NODE_ENV !== 'production' ? 'backdrop' : undefined
+          }
+          onClick={(event) => {
+            event.stopPropagation();
+            event.preventDefault();
+            if (onClose) {
+              handleOnClose({ closeTrigger: 'backdrop' });
+            }
+          }}
+          pointerEvents={!onClose ? 'none' : undefined}
+          position="fixed"
+          zIndex="modal"
+          top={0}
+          left={0}
+          className={styles.backdrop}
+        />
+
+        {/* Todo - use `popover` property when available */}
```

Comment:
I would prefer this is moved to our Browser Support "desired features list" reminding us to return here, rather than documenting at the call site.